File: examples/php/sort_2.php
<?php
function selection_sort($arr) {
    $n = count($arr);
    for ($i = 0; $i < $n - 1; $i++) {
        $min = $i;
        for ($j = $i + 1; $j < $n; $j++) {
            if ($arr[$j] < $arr[$min]) $min = $j;
        }
        if ($min !== $i) list($arr[$i], $arr[$min]) = [$arr[$min], $arr[$i]];
    }
    return $arr;
}
$data = [64, 34, 25, 12, 22];
echo implode(',', selection_sort($data)) . "\n";

File: examples/rust/sort_2.rs
fn insertion_sort(arr: &mut [i32]) {
    for i in 1..arr.len() {
        let key = arr[i];
        let mut j = i;
        while j > 0 && arr[j - 1] > key {
            arr[j] = arr[j - 1];
            j -= 1;
        }
        arr[j] = key;
    }
}

fn main() {
    let mut data = [64, 34, 25, 12, 22];
    insertion_sort(&mut data);
    println!("{:?}", data);
}
